crates/factor-key-value-spin/src/lib.rs

```diff
@@ -11,12 +11,16 @@ use spin_key_value_sqlite::{DatabaseLocation, KeyValueSqlite};
 /// A key-value store that uses SQLite as the backend.
 pub struct SpinKeyValueStore {
     /// The base path or directory for the SQLite database file.
-    base_path: PathBuf,
+    base_path: Option<PathBuf>,
 }
 
 impl SpinKeyValueStore {
     /// Create a new SpinKeyValueStore with the given base path.
-    pub fn new(base_path: PathBuf) -> Self {
+    ///
+    /// If the database directory is None, the database will always be in-memory.
```

Comment:
I think this isn't quite right. If the directory is None and there is an _absolute_ path I would expect that to work fine. This might be important for SpinKube.

When the directory is None and the path is _relative_ I'm not sure what the current behavior is but it seems like it would be least surprising for that to be a fatal error.

Reply:
I've changed it so that absolute paths are still allowed when `--state-dir` is unset. However, this does mean a runtime can never fully force in-memory databases.

Reply:
Ok I'll error for now - probably easier in the future to make it less restrictive than it is to make it more.

Reply:
> a runtime can never fully force in-memory databases

If I'm reading `main` correctly it looks like you can omit `path` to force an in-memory database.

Reply:
Yes, but that's in the runtime config not in the actual running of the runtime engine. Are we assuming that whoever is controlling the runtime is the same as whoever is setting the runtime config? We're requiring coordination between whoever is setting the runtime config and whoever is actually running the runtime binary.